### DISCUSSION TITLE
How do I connect to a local database container?

### DISCUSSION THREAD

**Discussion by azhi98** [microsoft/aspire#3143] [Q&A]
I have a running data container that I want to connect directly to and join the monitor.
`var mysqldb = builder.AddMySql("mysql").AddDatabase("mysqldb");` This code creates a new database container, which is not what I wanted.

**Comment by davidfowl** [comment]
Use `AddConnectionString` https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/external-parameters#connection-string-values

**Reply by azhi98** [reply]
Connecting to the local database is no way to join the monitor? It won't appear in "Resources".

**Reply by davidfowl** [reply]
What would show up there?

**Reply by azhi98** [reply]
Isn't there A way to connect to the local database and show up in the "Resources" panel like `var mysqldb = builder.AddMySql("mysql").AddDatabase("mysqldb");`?

**Reply by davidfowl** [reply]
That's a connection string resource (connect to an existing thing), not a database resource (which in this case means create a container).

**Reply by azhi98** [reply]
I know, I looked at the built-in resource types, and I wanted to add existing database resources to the resource as if I were creating a container. I might need to look at the source code for "AddContainer"

**Reply by azhi98** [reply]
Finally, I used a volume to store the data, and fixed the port and password so that I could view the data in the database through the connection tool.

**Reply by davidfowl** [reply]
So you are using the aspire to create the container now right?

**Reply by azhi98** [reply]
Yes, I am trying to develop with 'aspire'. Both 'mysql' and 'redis' in the project use data volumes created by 'aspire'.
